server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionResponse.java

```diff
@@ -0,0 +1,116 @@
+    private static final ParseField SKIP_UNAVAILABLE_FIELD = new ParseField("skip_unavailable");
+    private static final ParseField MATCHING_INDICES_FIELD = new ParseField("matching_indices");
+    private static final ParseField ES_VERSION_FIELD = new ParseField("version");
+    private static final ParseField ERROR_FIELD = new ParseField("error");
+
+    private final Map<String, ResolveClusterInfo> infoMap;
+
+    public ResolveClusterActionResponse(Map<String, ResolveClusterInfo> infoMap) {
+        this.infoMap = infoMap;
+    }
+
+    public ResolveClusterActionResponse(StreamInput in) throws IOException {
+        super(in);
+        if (in.getTransportVersion().before(TransportVersions.RESOLVE_CLUSTER_ENDPOINT_ADDED)) {
+            throw new UnsupportedOperationException(
+                "ResolveClusterAction requires at least Transport Version "
+                    + TransportVersions.RESOLVE_CLUSTER_ENDPOINT_ADDED
+                    + " but was "
+                    + in.getTransportVersion()
+            );
+        }
+        this.infoMap = in.readImmutableMap(ResolveClusterInfo::new);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().before(TransportVersions.RESOLVE_CLUSTER_ENDPOINT_ADDED)) {
+            throw new UnsupportedOperationException(
+                "ResolveClusterAction requires at least Transport Version "
+                    + TransportVersions.RESOLVE_CLUSTER_ENDPOINT_ADDED
+                    + " but was "
+                    + out.getTransportVersion()
+            );
+        }
+        out.writeMap(infoMap, StreamOutput::writeWriteable);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        for (Map.Entry<String, ResolveClusterInfo> entry : infoMap.entrySet()) {
+            String clusterAlias = entry.getKey();
+            if (clusterAlias.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)) {
+                clusterAlias = SearchResponse.LOCAL_CLUSTER_NAME_REPRESENTATION;
+            }
+            builder.startObject(clusterAlias);
+            ResolveClusterInfo clusterInfo = entry.getValue();
+            builder.field(CONNECTED_FIELD.getPreferredName(), clusterInfo.isConnected());
+            builder.field(SKIP_UNAVAILABLE_FIELD.getPreferredName(), clusterInfo.getSkipUnavailable());
+            if (clusterInfo.getError() != null) {
+                builder.field(ERROR_FIELD.getPreferredName(), clusterInfo.getError());
+            }
+            if (clusterInfo.getMatchingIndices() != null) {
+                builder.field(MATCHING_INDICES_FIELD.getPreferredName(), clusterInfo.getMatchingIndices());
+            }
+            Build build = clusterInfo.getBuild();
+            if (build != null) {
+                // TODO Lucene version is not part of build - do we want that as well?
```

Comment:
I don't think that is needed, given how the API is used?

Reply:
Thanks. Removed TODO.